### PR TITLE
refactor: eliminate `Ast.Constant.Regex` in `Simplifier`

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes https://github.com/flix/flix/issues/6047

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala [hunk 1]
```diff
@@ -116,11 +116,7 @@ object GenExpression {
         addSourceLine(mv, loc)
         mv.visitLdcInsn(s)
 
-      case Ast.Constant.Regex(patt) =>
-        addSourceLine(mv, loc)
-        mv.visitLdcInsn(patt.pattern)
-        mv.visitMethodInsn(INVOKESTATIC, JvmName.Regex.toInternalName, "compile",
-          AsmOps.getMethodDescriptor(List(JvmType.String), JvmType.Regex), false)
+      case _ => throw InternalCompilerException(s"Unexpected Constant $cst", loc)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala:119]: Can we not remove it from `Ast.Constant`? or is that not possible?

At minimum, I think we should replace `_` by matching on the Regex.

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala:119]: > Can we not remove it from `Ast.Constant`? or is that not possible?

Remove what exactly? :)

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala:119]: But can we drop `Ast.Constant.Regex` or is that used earlier in the compiler?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala:119]: I see its used earlier.

**Comment by magnus-madsen** [comment]
In that case I am not sure if this PR is worth it. Let me think.

**Comment by magnus-madsen** [comment]
I decided its not worth it. Sorry for the trouble.

**Comment by jaschdoc** [comment]
No worries :)

**Pull request closed without merging**